client/src/legacy/GridField.js

```diff
@@ -14,12 +12,9 @@ $.entwine('ss', function($) {
     onmatch: function () {
       if (this.needsColumnFix()) {
         this.fixColumns();
-        this.injectSearchButton(false);
       }
 
-      if (this.hasFilters()) {
-        this.injectSearchButton(true);
-      };
+      this.fixShowFilters();
```

Comment:
The search button isn't "injected" anymore - we rely on templates on page load instead. But we do sometimes need to fix the css selector due to the weird way gridfield refreshes sometimes.